lib/widgets/new_dm_sheet.dart

```diff
@@ -178,8 +181,13 @@ class _NewDmHeader extends StatelessWidget {
         final narrow = DmNarrow.withUsers(
           selectedUserIds.toList(),
           selfUserId: store.selfUserId);
-        Navigator.pushReplacement(context,
-          MessageListPage.buildRoute(context: context, narrow: narrow));
+        if (onSelect != null) {
+          Navigator.pop(context); // TODO why doesn't this work?
+          onSelect!(narrow);
+        } else {
+          Navigator.pushReplacement(context,
+            MessageListPage.buildRoute(context: context, narrow: narrow));
+        }
```

Comment:
Can this `else` case instead move to the caller that uses it? Then make `onSelect` non-nullable and just call it unconditionally.

The call could pass the context:
```dart
onSelect(context, narrow);
```
if that helps make that work.

Reply:
This is an example of a change I'd recommend writing as its own NFC prep commit, coming before the main commit: just introduce this onSelect field, and move this existing logic to the ultimate caller, without yet changing any behavior. Then a later commit adds a new caller that passes its own, different, `onSelect`.